### PR TITLE
diff-pr: ADD/COPY enumberate full list of options

### PR DESCRIPTION
**Pull request by grooverdan**
Buildkit has many more options on ADD/COPY than just chown

Lets exclude all of them.

ref:
* https://docs.docker.com/reference/dockerfile/#add
* https://docs.docker.com/reference/dockerfile/#copy

### CODE REVIEW

**Review by tianon — DISMISSED** [review]
This is a little bit different from the implementation in https://github.com/docker-library/bashbrew/blob/10343d331c8ea6e3ed289c613deb681abeb03381/cmd/bashbrew/docker.go#L159-L169 (more specific, where that just ignores "everything" it doesn't recognize -- pros and cons both ways), but I'm fine with it.

I will make it clear here though, for future travelers finding this reference, updating our script to ignore all these things does _not_ mean we will accept image updates that use all these things, and they will be evaluated on a case-by-case basis (as usual). :heart:

**Comment by tianon** [comment]
```
       next                  Stop processing the current input  record.   Read
                             the  next  input record and start processing over
                             with the first pattern in the AWK program.   Upon
                             reaching  the  end of the input data, execute any
                             END rule(s).
```

oops -- most of these should be just "ignore this flag" where invoke `next` means we ignore the entire `COPY` :innocent:

**Comment by grooverdan** [comment]
Opps indeed. - s/next/continue/ ?

**Comment by tianon** [comment]
Well, _almost_ -- the original code used `next` specifically because `--from` means we _do_ want to ignore the whole `COPY` line :sweat_smile: (that means it's a file from another image, not a file from the build context, which is what this code is trying to determine)

Also, if we want to be more strict, we need to stop ignoring `--*` arguments after the first non-`--` argument (and assume everything else is an argument, not a flag, because they can't be interspersed, which is part of what the bashbrew implementation does differently that we've been kind of ignoring until now because our set of recognized flags was small :innocent:)

**Comment by grooverdan** [comment]
Done. Even did some testing this time :grimacing: . And --from is truely a `next` option.

**Review by tianon — APPROVED** [review]
Oh, very cute now, thank you! :+1:

**Comment by tianon** [comment]
I think we've got a fencepost bug:

```
cp: cannot stat 'tar/mariadb_10.4//usr/local/bin/healthcheck.sh': No such file or directory
```

(after running `./diff-pr.sh 16944`)

File: diff-pr.sh [hunk 1]
```diff
@@ -249,7 +249,17 @@ copy-tar() {
...
+						continue
+					}
+					# additional ADD options
+					if ($i ~ /^--(keep-git-dir|checksum)=/) {
+						continue
+					}
+					print $i
+					while (i < NF) {
+						i++
 						print $i
```
Comment (tianon) [inline diff-pr.sh:263]: Maybe we just embed the same c-style loop as above here?

```suggestion
					for ( ; i < NF; i++) {
						print $i
```

Comment (tianon) [reply diff-pr.sh:263]: Verified functional :eyes:

<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-cat b/_bashbrew-cat
index 0921c28..5a839ca 100644
--- a/_bashbrew-cat
+++ b/_bashbrew-cat
@@ -1,62 +1,63 @@
 Maintainers: Daniel Black <daniel@mariadb.org> (@grooverdan), Daniel Bartholomew <dbart@mariadb.com> (@dbart), Faustin Lammler <faustin@mariadb.org> (@fauust)
 GitRepo: https://github.com/MariaDB/mariadb-docker.git
+Builder: buildkit
 
 Tags: 10.4.34-focal, 10.4-focal, 10.4.34, 10.4
 Architectures: amd64, arm64v8, ppc64le
-GitCommit: 44ed2e231207787c08d56acc94b79d5f06efe006
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 10.4
 
 Tags: 10.5.25-focal, 10.5-focal, 10.5.25, 10.5
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: 44ed2e231207787c08d56acc94b79d5f06efe006
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 10.5
 
 Tags: 10.6.18-focal, 10.6-focal, 10.6.18, 10.6
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: 44ed2e231207787c08d56acc94b79d5f06efe006
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 10.6
 
 Tags: 10.6.18-ubi9, 10.6-ubi9, 10.6.18-ubi, 10.6-ubi
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: eac33f601ec635b4e0ce921efc6ac63884abceaf
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 10.6-ubi
 
 Tags: 10.11.8-jammy, 10.11-jammy, 10-jammy, 10.11.8, 10.11, 10
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: 44ed2e231207787c08d56acc94b79d5f06efe006
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 10.11
 
 Tags: 10.11.8-ubi9, 10.11-ubi9, 10-ubi9, 10.11.8-ubi, 10.11-ubi, 10-ubi
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: eac33f601ec635b4e0ce921efc6ac63884abceaf
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 10.11-ubi
 
 Tags: 11.1.5-jammy, 11.1-jammy, 11.1.5, 11.1
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: 44ed2e231207787c08d56acc94b79d5f06efe006
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 11.1
 
 Tags: 11.2.4-jammy, 11.2-jammy, 11.2.4, 11.2
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: 44ed2e231207787c08d56acc94b79d5f06efe006
+GitCommit: 81c4ce79659a9b705686b5704a3f3d3b9119532a
 Directory: 11.2
 
 Tags: 11.4.2-noble, 11.4-noble, 11-noble, noble, lts-noble, 11.4.2, 11.4, 11, latest, lts
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: 4e20774a56c8fb93cec9ee9d4a5b476bc0f8dd0d
+GitCommit: 11135d071fd1fe355b1f7fa99b9d3b4a59bb5225
 Directory: 11.4
 
 Tags: 11.4.2-ubi9, 11.4-ubi9, 11-ubi9, lts-ubi9, 11.4.2-ubi, 11.4-ubi, 11-ubi, lts-ubi
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: eac33f601ec635b4e0ce921efc6ac63884abceaf
+GitCommit: 11135d071fd1fe355b1f7fa99b9d3b4a59bb5225
 Directory: 11.4-ubi
 
 Tags: 11.5.1-noble-rc, 11.5-noble-rc, 11.5.1-rc, 11.5-rc
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: 7dabf47ac1c0f2f324563f5a0616812bf332f781
+GitCommit: 11135d071fd1fe355b1f7fa99b9d3b4a59bb5225
 Directory: 11.5
 
 Tags: 11.5.1-ubi9-rc, 11.5-ubi9-rc, 11.5.1-ubi-rc, 11.5-ubi-rc
 Architectures: amd64, arm64v8, ppc64le, s390x
-GitCommit: eac33f601ec635b4e0ce921efc6ac63884abceaf
+GitCommit: 11135d071fd1fe355b1f7fa99b9d3b4a59bb5225
 Directory: 11.5-ubi
diff --git a/mariadb_10-ubi/Dockerfile b/mariadb_10-ubi/Dockerfile
index 0483d78..272ae5f 100644
--- a/mariadb_10-ubi/Dockerfile
+++ b/mariadb_10-ubi/Dockerfile
@@ -28,7 +28,7 @@ RUN set -eux; \
 	gosu --version; \
 	gosu nobody true
 
-COPY docker.cnf /etc/my.cnf.d/
+COPY --chmod=0644 docker.cnf /etc/my.cnf.d/
 
 COPY MariaDB.repo /etc/yum.repos.d/
 
diff --git a/mariadb_10-ubi/docker-entrypoint.sh b/mariadb_10-ubi/docker-entrypoint.sh
index 6e50e1c..4af730a 100755
--- a/mariadb_10-ubi/docker-entrypoint.sh
+++ b/mariadb_10-ubi/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_10-ubi/docker.cnf b/mariadb_10-ubi/docker.cnf
index eb426d0..978fe6d 100644
--- a/mariadb_10-ubi/docker.cnf
+++ b/mariadb_10-ubi/docker.cnf
@@ -4,6 +4,11 @@
 host-cache-size=0
 skip-name-resolve
 
+expire_logs_days=10
+character-set-server=utf8mb4
+
+collation-server=utf8mb4_general_ci # 10*
+
 [client-server]
 socket=/run/mariadb/mariadb.sock
 
diff --git a/mariadb_10.4/docker-entrypoint.sh b/mariadb_10.4/docker-entrypoint.sh
index 85110e8..ab540da 100755
--- a/mariadb_10.4/docker-entrypoint.sh
+++ b/mariadb_10.4/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_10.5/docker-entrypoint.sh b/mariadb_10.5/docker-entrypoint.sh
index f6c5dd5..07e1ab4 100755
--- a/mariadb_10.5/docker-entrypoint.sh
+++ b/mariadb_10.5/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_10.6-ubi/Dockerfile b/mariadb_10.6-ubi/Dockerfile
index d4dc725..eaf6b17 100644
--- a/mariadb_10.6-ubi/Dockerfile
+++ b/mariadb_10.6-ubi/Dockerfile
@@ -28,7 +28,7 @@ RUN set -eux; \
 	gosu --version; \
 	gosu nobody true
 
-COPY docker.cnf /etc/my.cnf.d/
+COPY --chmod=0644 docker.cnf /etc/my.cnf.d/
 
 COPY MariaDB.repo /etc/yum.repos.d/
 
diff --git a/mariadb_10.6-ubi/docker-entrypoint.sh b/mariadb_10.6-ubi/docker-entrypoint.sh
index 4738c6a..fb0cc6c 100755
--- a/mariadb_10.6-ubi/docker-entrypoint.sh
+++ b/mariadb_10.6-ubi/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_10.6-ubi/docker.cnf b/mariadb_10.6-ubi/docker.cnf
index eb426d0..978fe6d 100644
--- a/mariadb_10.6-ubi/docker.cnf
+++ b/mariadb_10.6-ubi/docker.cnf
@@ -4,6 +4,11 @@
 host-cache-size=0
 skip-name-resolve
 
+expire_logs_days=10
+character-set-server=utf8mb4
+
+collation-server=utf8mb4_general_ci # 10*
+
 [client-server]
 socket=/run/mariadb/mariadb.sock
 
diff --git a/mariadb_10.6/docker-entrypoint.sh b/mariadb_10.6/docker-entrypoint.sh
index 4738c6a..fb0cc6c 100755
--- a/mariadb_10.6/docker-entrypoint.sh
+++ b/mariadb_10.6/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_10/docker-entrypoint.sh b/mariadb_10/docker-entrypoint.sh
index 6e50e1c..4af730a 100755
--- a/mariadb_10/docker-entrypoint.sh
+++ b/mariadb_10/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_11.1/docker-entrypoint.sh b/mariadb_11.1/docker-entrypoint.sh
index 8e15ea0..1495de6 100755
--- a/mariadb_11.1/docker-entrypoint.sh
+++ b/mariadb_11.1/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_11.2/docker-entrypoint.sh b/mariadb_11.2/docker-entrypoint.sh
index e41ed9c..5e89cc7 100755
--- a/mariadb_11.2/docker-entrypoint.sh
+++ b/mariadb_11.2/docker-entrypoint.sh
@@ -121,7 +121,8 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +134,8 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_11.5-rc/Dockerfile b/mariadb_11.5-rc/Dockerfile
index 7ec7d93..f70beff 100644
--- a/mariadb_11.5-rc/Dockerfile
+++ b/mariadb_11.5-rc/Dockerfile
@@ -82,7 +82,7 @@ LABEL org.opencontainers.image.authors="MariaDB Community" \
 ARG MARIADB_VERSION=1:11.5.1+maria~ubu2404
 ENV MARIADB_VERSION $MARIADB_VERSION
 # release-status:RC
-# release-support-type:Short Term Support
+# release-support-type:Rolling
 # (https://downloads.mariadb.org/rest-api/mariadb/)
 
 # Allowing overriding of REPOSITORY, a URL that includes suite and component for testing and Enterprise Versions
diff --git a/mariadb_11.5-rc/docker-entrypoint.sh b/mariadb_11.5-rc/docker-entrypoint.sh
index d8d705a..fa35d57 100755
--- a/mariadb_11.5-rc/docker-entrypoint.sh
+++ b/mariadb_11.5-rc/docker-entrypoint.sh
@@ -121,7 +121,9 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		--skip-ssl --ssl-cert='' --ssl-key='' --ssl-ca='' \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +135,9 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			--skip-ssl --skip-ssl-verify-server-cert \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_11.5-rc/healthcheck.sh b/mariadb_11.5-rc/healthcheck.sh
index 784f9bd..9138c77 100755
--- a/mariadb_11.5-rc/healthcheck.sh
+++ b/mariadb_11.5-rc/healthcheck.sh
@@ -42,6 +42,7 @@ _process_sql()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}} \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}} \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-B "$@"
 }
 
@@ -62,6 +63,7 @@ connect()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}}  \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}}  \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-h localhost --protocol tcp -e 'select 1' 2>&1 \
 		| grep -qF "Can't connect"
 	local ret=${PIPESTATUS[1]}
diff --git a/mariadb_11.5-ubi-rc/Dockerfile b/mariadb_11.5-ubi-rc/Dockerfile
index f9e272e..ffc3aeb 100644
--- a/mariadb_11.5-ubi-rc/Dockerfile
+++ b/mariadb_11.5-ubi-rc/Dockerfile
@@ -28,7 +28,7 @@ RUN set -eux; \
 	gosu --version; \
 	gosu nobody true
 
-COPY docker.cnf /etc/my.cnf.d/
+COPY --chmod=0644 docker.cnf /etc/my.cnf.d/
 
 COPY MariaDB.repo /etc/yum.repos.d/
 
@@ -56,7 +56,7 @@ LABEL org.opencontainers.image.authors="MariaDB Community" \
 # bashbrew-architectures: amd64 arm64v8 ppc64le s390x
 ARG MARIADB_VERSION=11.5.1
 # release-status:RC
-# release-support-type:Short Term Support
+# release-support-type:Rolling
 # (https://downloads.mariadb.org/rest-api/mariadb/)
 
 # missing pwgen(epel), jemalloc(epel) (as entrypoint/user extensions)
diff --git a/mariadb_11.5-ubi-rc/docker-entrypoint.sh b/mariadb_11.5-ubi-rc/docker-entrypoint.sh
index d8d705a..fa35d57 100755
--- a/mariadb_11.5-ubi-rc/docker-entrypoint.sh
+++ b/mariadb_11.5-ubi-rc/docker-entrypoint.sh
@@ -121,7 +121,9 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		--skip-ssl --ssl-cert='' --ssl-key='' --ssl-ca='' \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +135,9 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			--skip-ssl --skip-ssl-verify-server-cert \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_11.5-ubi-rc/docker.cnf b/mariadb_11.5-ubi-rc/docker.cnf
index eb426d0..f3af00a 100644
--- a/mariadb_11.5-ubi-rc/docker.cnf
+++ b/mariadb_11.5-ubi-rc/docker.cnf
@@ -4,6 +4,11 @@
 host-cache-size=0
 skip-name-resolve
 
+expire_logs_days=10
+character-set-server=utf8mb4
+
+character-set-collations=utf8mb4=uca1400_ai_ci # 11.3+
+
 [client-server]
 socket=/run/mariadb/mariadb.sock
 
diff --git a/mariadb_11.5-ubi-rc/healthcheck.sh b/mariadb_11.5-ubi-rc/healthcheck.sh
index 784f9bd..9138c77 100755
--- a/mariadb_11.5-ubi-rc/healthcheck.sh
+++ b/mariadb_11.5-ubi-rc/healthcheck.sh
@@ -42,6 +42,7 @@ _process_sql()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}} \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}} \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-B "$@"
 }
 
@@ -62,6 +63,7 @@ connect()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}}  \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}}  \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-h localhost --protocol tcp -e 'select 1' 2>&1 \
 		| grep -qF "Can't connect"
 	local ret=${PIPESTATUS[1]}
diff --git a/mariadb_lts-ubi/Dockerfile b/mariadb_lts-ubi/Dockerfile
index 6a367f7..6ec1021 100644
--- a/mariadb_lts-ubi/Dockerfile
+++ b/mariadb_lts-ubi/Dockerfile
@@ -28,7 +28,7 @@ RUN set -eux; \
 	gosu --version; \
 	gosu nobody true
 
-COPY docker.cnf /etc/my.cnf.d/
+COPY --chmod=0644 docker.cnf /etc/my.cnf.d/
 
 COPY MariaDB.repo /etc/yum.repos.d/
 
diff --git a/mariadb_lts-ubi/docker-entrypoint.sh b/mariadb_lts-ubi/docker-entrypoint.sh
index e5b4aa9..eb13c84 100755
--- a/mariadb_lts-ubi/docker-entrypoint.sh
+++ b/mariadb_lts-ubi/docker-entrypoint.sh
@@ -121,7 +121,9 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		--skip-ssl --ssl-cert='' --ssl-key='' --ssl-ca='' \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +135,9 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			--skip-ssl --skip-ssl-verify-server-cert \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_lts-ubi/docker.cnf b/mariadb_lts-ubi/docker.cnf
index eb426d0..f3af00a 100644
--- a/mariadb_lts-ubi/docker.cnf
+++ b/mariadb_lts-ubi/docker.cnf
@@ -4,6 +4,11 @@
 host-cache-size=0
 skip-name-resolve
 
+expire_logs_days=10
+character-set-server=utf8mb4
+
+character-set-collations=utf8mb4=uca1400_ai_ci # 11.3+
+
 [client-server]
 socket=/run/mariadb/mariadb.sock
 
diff --git a/mariadb_lts-ubi/healthcheck.sh b/mariadb_lts-ubi/healthcheck.sh
index 784f9bd..9138c77 100755
--- a/mariadb_lts-ubi/healthcheck.sh
+++ b/mariadb_lts-ubi/healthcheck.sh
@@ -42,6 +42,7 @@ _process_sql()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}} \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}} \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-B "$@"
 }
 
@@ -62,6 +63,7 @@ connect()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}}  \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}}  \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-h localhost --protocol tcp -e 'select 1' 2>&1 \
 		| grep -qF "Can't connect"
 	local ret=${PIPESTATUS[1]}
diff --git a/mariadb_lts/docker-entrypoint.sh b/mariadb_lts/docker-entrypoint.sh
index e5b4aa9..eb13c84 100755
--- a/mariadb_lts/docker-entrypoint.sh
+++ b/mariadb_lts/docker-entrypoint.sh
@@ -121,7 +121,9 @@ mysql_get_config() {
 docker_temp_server_start() {
 	"$@" --skip-networking --default-time-zone=SYSTEM --socket="${SOCKET}" --wsrep_on=OFF \
 		--expire-logs-days=0 \
-		--loose-innodb_buffer_pool_load_at_startup=0 &
+		--loose-innodb_buffer_pool_load_at_startup=0 \
+		--skip-ssl --ssl-cert='' --ssl-key='' --ssl-ca='' \
+		&
 	declare -g MARIADB_PID
 	MARIADB_PID=$!
 	mysql_note "Waiting for server startup"
@@ -133,7 +135,9 @@ docker_temp_server_start() {
 	fi
 	local i
 	for i in {30..0}; do
-		if docker_process_sql "${extraArgs[@]}" --database=mysql <<<'SELECT 1' &> /dev/null; then
+		if docker_process_sql "${extraArgs[@]}" --database=mysql \
+			--skip-ssl --skip-ssl-verify-server-cert \
+			<<<'SELECT 1' &> /dev/null; then
 			break
 		fi
 		sleep 1
diff --git a/mariadb_lts/healthcheck.sh b/mariadb_lts/healthcheck.sh
index 784f9bd..9138c77 100755
--- a/mariadb_lts/healthcheck.sh
+++ b/mariadb_lts/healthcheck.sh
@@ -42,6 +42,7 @@ _process_sql()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}} \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}} \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-B "$@"
 }
 
@@ -62,6 +63,7 @@ connect()
 		${def['file']:+--defaults-file=${def['file']}} \
 		${def['extra_file']:+--defaults-extra-file=${def['extra_file']}}  \
 		${def['group_suffix']:+--defaults-group-suffix=${def['group_suffix']}}  \
+		--skip-ssl --skip-ssl-verify-server-cert \
 		-h localhost --protocol tcp -e 'select 1' 2>&1 \
 		| grep -qF "Can't connect"
 	local ret=${PIPESTATUS[1]}
```

</details>

**Review by tianon — APPROVED** [review]

**Comment by tianon** [comment]
Success, thanks for sticking with it :smile:

**Pull request merged**